src/main/java/core/basesyntax/service/impl/ReturnFruitOperation.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitDto;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitStorage;
+import core.basesyntax.service.FruitOperations;
+import java.util.Map;
+
+public class ReturnFruitOperation implements FruitOperations {
+    private Map<String, Fruit> storage = FruitStorage.getFruitStorage();
+
+    @Override
+    public void doOperation(FruitDto fruitDto) {
+        String fruitName = fruitDto.getFruitName();
+        Integer amountToReturn = fruitDto.getAmount();
+        Fruit fruit;
+        if (storage.containsKey(fruitName)) {
+            fruit = new SupplyFruitOperation().getFruit(fruitDto, fruitName, amountToReturn);
```

Comment:
it is not very good that we are calling `new SupplyFruitOperation().getFruit()` here

let's refactor `FruitOperation` implementations

create a class `StorageService` where you will have all methods that communicate with storage (add(), remove() etc.) including helper methods

then in your `FruitOperation` implementations create  `StorageService` field and call its methods inside `doOperation()`